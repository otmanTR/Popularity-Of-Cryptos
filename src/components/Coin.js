import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Coin = ({
  name, price, symbol, image, id,
}) => (
  <Link to={`/coin/${id}`} className="link">
    <li className="coin">
      <img className="coinImage" src={image} alt={image} />
      <h2 className="coinName">
        {name}
        <br />
        {symbol}
      </h2>
      <p>
        {price}
        $
      </p>
    </li>
  </Link>
);

Coin.propTypes = {
  name: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  symbol: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
};

export default Coin;
