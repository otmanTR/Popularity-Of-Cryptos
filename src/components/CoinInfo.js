import PropTypes from 'prop-types';
import React from 'react';

const CoinInfo = ({
  image,
  price,
  name,
  symbol,
  marketCap,
  marketCapRank,
  totalVolume,
  high24h,
  low24h,
  priceChange24h,
  marketCapChange24h,
  totalSupply,
}) => (
  <>
    <div>
      <div>
        <img className="coinImage" src={image} alt={image} />
      </div>
      <div className="infoText">
        <p>Crypto Coin Info:</p>
      </div>
      <ul className="detailUl">
        <li className="detail">
          Name:
          {name}
        </li>
        <li className="detail">
          Symbol:
          {symbol}
        </li>
        <li className="detail">
          Price:
          {price}
          $
        </li>
        <li className="detail">
          Market Cap:
          {marketCap}
          $
        </li>
        <li className="detail">
          Market Cap Rank:
          {marketCapRank}
          st.
        </li>
        <li className="detail">
          Total Volume:
          {totalVolume}
          $
        </li>
        <li className="detail">
          Highest 24h:
          {high24h}
          $
        </li>
        <li className="detail">
          Lowest 24h
          {low24h}
          $
        </li>
        <li className="detail">
          Price Change 24h:
          {priceChange24h}
          $
        </li>
        <li className="detail">
          Market Cap Change:
          {marketCapChange24h}
          $
        </li>
        <li className="detail">
          Total Supply:
          {totalSupply}
        </li>
      </ul>
    </div>
  </>
);

CoinInfo.propTypes = {
  marketCap: PropTypes.string.isRequired,
  marketCapRank: PropTypes.string.isRequired,
  totalVolume: PropTypes.string.isRequired,
  high24h: PropTypes.string.isRequired,
  low24h: PropTypes.string.isRequired,
  priceChange24h: PropTypes.string.isRequired,
  marketCapChange24h: PropTypes.string.isRequired,
  totalSupply: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  symbol: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};

export default CoinInfo;
