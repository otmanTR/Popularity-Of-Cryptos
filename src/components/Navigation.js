import { FaChevronLeft } from 'react-icons/fa';
import { Link } from 'react-router-dom';

export const NavBar = () => (
  <>
    <div className="navContainer">
      <div>
        <Link to="/" className="home-btn">
          <FaChevronLeft />
        </Link>
      </div>
      <div className="year"> 2023</div>
    </div>
  </>
);

export default { NavBar };
