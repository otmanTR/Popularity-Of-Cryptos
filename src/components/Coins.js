import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getCoins } from '../redux/coins/coinsSlice';
import Coin from './Coin';

const Coins = () => {
  const dispatch = useDispatch();
  const coins = useSelector((state) => state.coins);
  // console.log(coin)
  useEffect(() => {
    if (coins.length === 0) {
      dispatch(getCoins());
    }
  }, [coins, dispatch]);

  return (

    <div className="main">

      <h1 className="title">Crypto Coins</h1>

      <ul className="homeCoinList">
        {coins.map((coin) => (
          <Coin
            key={coin.name}
            id={coin.id}
            name={coin.name}
            price={coin.price}
            symbol={coin.symbol}
            image={coin.image}
          />
        ))}
      </ul>
    </div>
  );
};

export default Coins;
