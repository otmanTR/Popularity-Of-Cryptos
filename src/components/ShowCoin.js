import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import CoinInfo from './CoinInfo';
import { getCoins } from '../redux/coins/coinsSlice';

const ShowCoin = () => {
  const coins = useSelector((state) => state.coins);
  const dispatch = useDispatch();
  const { id } = useParams();

  useEffect(() => {
    if (coins.length === 0) {
      dispatch(getCoins());
    }
  }, [coins, dispatch]);

  const oneCoin = coins.filter((coin) => coin.id === id);

  return (
    <ul className="details-ul">
      {oneCoin && oneCoin.map((coin) => (
        <CoinInfo
          key={coin.id}
          image={coin.image}
          price={coin.price}
          name={coin.name}
          symbol={coin.symbol}
          marketCap={coin.symbol}
          marketCapRank={coin.marketCap}
          totalVolume={coin.totalVolume}
          high24h={coin.high24h}
          low24h={coin.low24h}
          priceChange24h={coin.priceChange24h}
          marketCapChange24h={coin.marketCapChange24h}
          marketCapChangePercentage24h={coin.marketCapChangePercentage24h}
          totalSupply={coin.totalSupply}
        />
      ))}
    </ul>
  );
};

export default ShowCoin;
