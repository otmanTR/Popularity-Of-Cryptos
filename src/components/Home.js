import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { FaSearch } from 'react-icons/fa';
import { getCoins } from '../redux/coins/coinsSlice';
import Coins from './Coins';
import { NavBar } from './Navigation';

export const Home = () => {
  const coin = useSelector((state) => state.coins);
  const dispatch = useDispatch();
  const [search, setSearch] = useState('');

  useEffect(() => {
    dispatch(getCoins());
  }, [dispatch]);

  return (
    <div className="mainContainer">
      <div>
        <h1 className="mainTitle">Choose Your Coin</h1>
        <div className="container">
          <div>
            <NavBar />
          </div>
          <div className="searchBox">
            <input
              type="text"
              className="searchInput"
              placeholder="Search..."
              onChange={(event) => setSearch(event.target.value)}
            />
          </div>
          <div>
            <button aria-label="Mute volume" type="button" className="searchButton" onChange={(e) => setSearch(e.target.value)}>
              <FaSearch />
            </button>
          </div>
        </div>
      </div>
      <div className="coinItems">
        <Coins coins={coin.filter((coin) => (search.toLowerCase() === '' ? coin : coin.name.toLowerCase().includes(search.toLowerCase())))} />

      </div>
    </div>
  );
};

export default { Home };
