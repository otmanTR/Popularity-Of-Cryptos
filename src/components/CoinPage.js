import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import ShowCoin from './ShowCoin';
import { getCoins } from '../redux/coins/coinsSlice';

const CoinPage = () => {
  const coins = useSelector((state) => state.coins);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getCoins());
  }, [dispatch]);

  return (
    <div className="coin-items">
      <div className="main-container-details">
        <a className="back-pointer" href="/">
          <i className="fa-solid fa-left-long" />
        </a>
        <h2 className="details-sec">Details</h2>
      </div>
      <ShowCoin coins={coins} />
    </div>
  );
};

export default { CoinPage };
