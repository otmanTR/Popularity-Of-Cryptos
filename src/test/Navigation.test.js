import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { NavBar } from '../components/Navigation';

describe('NavBar Component', () => {
  test('component match with the snapshot', () => {
    const tree = renderer.create(<BrowserRouter><NavBar /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
