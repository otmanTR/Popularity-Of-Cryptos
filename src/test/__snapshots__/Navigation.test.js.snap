// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin Card Component component match with the snapshot 1`] = `
<a
  className="link"
  href="/coin/undefined"
  onClick={[Function]}
>
  <li
    className="coin"
  >
    <img
      className="coinImage"
    />
    <h2
      className="coinName"
    >
      <br />
    </h2>
    <p
      className="coinPrice"
    >
      price:
      $
    </p>
  </li>
</a>
`;

exports[`Nav Component component match with the snapshot 1`] = `
<div>
  <a
    className="home-btn"
    href="/"
    onClick={[Function]}
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 320 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
      />
    </svg>
  </a>
</div>
`;

exports[`NavBar Component component match with the snapshot 1`] = `
<div>
  <div>
    <a
      className="home-btn"
      href="/"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 320 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
        />
      </svg>
    </a>
  </div>
  <div
    className="year"
  >
     2023
  </div>
  <div
    className="title"
  >
    Crypto Coins
  </div>
  <div
    className="searchBox"
  >
    <input
      className="searchInput"
      onChange={[Function]}
      placeholder="Search..."
      type="text"
    />
  </div>
  <div>
    <button
      aria-label="Mute volume"
      className="searchButton"
      onChange={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
        />
      </svg>
    </button>
  </div>
</div>
`;
